### ISSUE TITLE
Create BrancBlock for "case null" sections to avoid exceptions in the exploded graph

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-csharp#1801]
`case null` should generate BranchBlock instead of BinaryBranchBlock when inside a `switch` block with patterns.

The currently created BinaryBranchBlock causes an exception to be thrown when it is visited by the ExplodedGraph. The exception is thrown because the BinaryBranchBlock does not contain an instruction for the branch condition. Such branch condition would be difficult to be created because we cannot easily put more complex expressions in the CFG:
```
switch (a ?? b)
{
	case null:
		break;
}
```

Using a simple `BranchBlock` will generate some false negatives, but will work without exceptions and will be generally consistent with the other pattern sections (they don't apply constraints on the switch expression, but only on the declared variable).

Example False Negative for the condition evaluates to constant rule:

```
switch (o)
{
	case null:
		if (o == null) // False Negative, we don't know anything for "o" at this point
		{
		}
		break;
}
```
```
switch (o)
{
	case null:
		return o.ToString(); // False Negative, we don't know anything for "o" at this point
}
```
